src/components/WizardGroup.tsx

```diff
@@ -0,0 +1,58 @@
+import { handleDispatchWizard } from '../events/wizardActions';
+import { Tile } from '../systems/world_generation/Tile';
+import { Wizard } from '../types';
+import { WizardProfile } from './WizardProfile';
+import { useEffect } from 'preact/hooks';
+
+type WizardGroupProps = {
+  wizardRow: Wizard[];
+  keybinds?: string[];
+  dispatchToTile?: Tile;
+};
+
+export const WizardGroup: React.FC<WizardGroupProps> = (props) => {
+  const wizards = props.wizardRow;
+
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (!props.keybinds) return;
+      const keyIndex = props.keybinds.indexOf(event.key);
+      if (keyIndex !== -1) {
+        const wizard = wizards[keyIndex];
+        if (wizard && props.dispatchToTile) {
+          handleDispatchWizard(wizard, props.dispatchToTile);
+        }
+      }
+    };
+
+    window.addEventListener('keydown', handleKeyDown);
+
+    return () => {
+      window.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [props.keybinds, wizards]);
```

Comment:
Could potentially be simplified with https://usehooks-ts.com/react-hook/use-event-listener - it does the adding/removing in the hook so it should be a fairly minimal implementation